diff --git a/docs/wiki/Configuration:-Outputs.md b/docs/wiki/Configuration:-Outputs.md
index 73a02201..cd68ec66 100644
--- a/docs/wiki/Configuration:-Outputs.md
+++ b/docs/wiki/Configuration:-Outputs.md
@@ -15,6 +15,7 @@ output "eDP-1" {
     variable-refresh-rate // on-demand=true
     focus-at-startup
     backdrop-color "#001100"
+    bpc 10
 
     hot-corners {
         // off
@@ -279,6 +280,21 @@ output "HDMI-A-1" {
 }
 ```
 
+### `bpc`
+
+<sup>Since: next release</sup>
+
+Set the maximum bits per channel (BPC) for this output.
+This is not needed for most outputs unless encountering bandwidth or lower-than-expected colour depth issues.
+
+Valid values are `8`, `10`.
+
+```kdl
+output "HDMI-A-1" {
+    bpc 10
+}
+```
+
 ### `hot-corners`
 
 <sup>Since: 25.11</sup>
diff --git a/niri-config/src/lib.rs b/niri-config/src/lib.rs
index 329b7669..133361f7 100644
--- a/niri-config/src/lib.rs
+++ b/niri-config/src/lib.rs
@@ -744,6 +744,7 @@ mod tests {
                 transform "flipped-90"
                 position x=10 y=20
                 mode "1920x1080@144"
+                bpc 10
                 variable-refresh-rate on-demand=true
                 background-color "rgba(25, 25, 102, 1.0)"
                 hot-corners {
@@ -856,7 +857,7 @@ mod tests {
                 window-open { off; }
 
                 window-close {
-                    curve "cubic-bezier" 0.05 0.7 0.1 1  
+                    curve "cubic-bezier" 0.05 0.7 0.1 1
                 }
 
                 recent-windows-close {
@@ -1158,6 +1159,9 @@ mod tests {
                                 y: 20,
                             },
                         ),
+                        bpc: Some(
+                            _10,
+                        ),
                         mode: Some(
                             Mode {
                                 custom: false,
@@ -1203,6 +1207,7 @@ mod tests {
                         scale: None,
                         transform: Normal,
                         position: None,
+                        bpc: None,
                         mode: Some(
                             Mode {
                                 custom: true,
@@ -1229,6 +1234,7 @@ mod tests {
                         scale: None,
                         transform: Normal,
                         position: None,
+                        bpc: None,
                         mode: None,
                         modeline: Some(
                             Modeline {
diff --git a/niri-config/src/output.rs b/niri-config/src/output.rs
index 5d6565f5..f9fb0cec 100644
--- a/niri-config/src/output.rs
+++ b/niri-config/src/output.rs
@@ -59,6 +59,8 @@ pub struct Output {
     pub transform: Transform,
     #[knuffel(child)]
     pub position: Option<Position>,
+    #[knuffel(child, unwrap(argument))]
+    pub bpc: Option<Bpc>,
     #[knuffel(child)]
     pub mode: Option<Mode>,
     #[knuffel(child)]
@@ -101,6 +103,7 @@ impl Default for Output {
             scale: None,
             transform: Transform::Normal,
             position: None,
+            bpc: None,
             mode: None,
             modeline: None,
             variable_refresh_rate: None,
@@ -128,6 +131,13 @@ pub struct Position {
     pub y: i32,
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+pub enum Bpc {
+    _8 = 8,
+    #[default]
+    _10 = 10,
+}
+
 #[derive(knuffel::Decode, Debug, Clone, PartialEq, Default)]
 pub struct Vrr {
     #[knuffel(property, default = false)]
@@ -257,6 +267,52 @@ impl OutputName {
     }
 }
 
+impl TryFrom<u8> for Bpc {
+    type Error = &'static str;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            8 => Ok(Bpc::_8),
+            10 => Ok(Bpc::_10),
+            _ => Err(r#"invalid bpc, can be 8 or 10"#),
+        }
+    }
+}
+
+impl<S: ErrorSpan> knuffel::DecodeScalar<S> for Bpc {
+    fn type_check(
+        type_name: &Option<knuffel::span::Spanned<knuffel::ast::TypeName, S>>,
+        ctx: &mut Context<S>,
+    ) {
+        if let Some(type_name) = &type_name {
+            ctx.emit_error(DecodeError::unexpected(
+                type_name,
+                "type name",
+                "no type name expected for this node",
+            ));
+        }
+    }
+
+    fn raw_decode(
+        value: &knuffel::span::Spanned<knuffel::ast::Literal, S>,
+        ctx: &mut Context<S>,
+    ) -> Result<Self, DecodeError<S>> {
+        match &**value {
+            knuffel::ast::Literal::Int(ref val) => match u8::try_from(val) {
+                Ok(v) => Bpc::try_from(v).map_err(|e| DecodeError::conversion(value, e)),
+                Err(e) => {
+                    ctx.emit_error(DecodeError::conversion(value, e));
+                    Ok(Self::default())
+                }
+            },
+            _ => {
+                ctx.emit_error(DecodeError::scalar_kind(knuffel::decode::Kind::Int, value));
+                Ok(Self::default())
+            }
+        }
+    }
+}
+
 impl<S: ErrorSpan> knuffel::Decode<S> for Mode {
     fn decode_node(node: &SpannedNode<S>, ctx: &mut Context<S>) -> Result<Self, DecodeError<S>> {
         if let Some(type_name) = &node.type_name {
diff --git a/niri-ipc/src/lib.rs b/niri-ipc/src/lib.rs
index 0aa3cb4f..a5f61876 100644
--- a/niri-ipc/src/lib.rs
+++ b/niri-ipc/src/lib.rs
@@ -1228,6 +1228,10 @@ pub struct Output {
     ///
     /// `None` if the output is not mapped to any logical output (for example, if it is disabled).
     pub logical: Option<LogicalOutput>,
+    /// Maximum bit per color  (bit depth), if known.
+    pub max_bpc: Option<u8>,
+    /// Pixel format for framebuffers, if known.
+    pub format: Option<String>,
 }
 
 /// Output mode.
diff --git a/src/backend/headless.rs b/src/backend/headless.rs
index 8b5ee827..031f2cd2 100644
--- a/src/backend/headless.rs
+++ b/src/backend/headless.rs
@@ -109,6 +109,8 @@ impl Headless {
                 vrr_supported: false,
                 vrr_enabled: false,
                 logical: Some(logical_output(&output)),
+                max_bpc: None,
+                format: None,
             },
         );
 
diff --git a/src/backend/tty.rs b/src/backend/tty.rs
index 5368c7ec..913380fb 100644
--- a/src/backend/tty.rs
+++ b/src/backend/tty.rs
@@ -14,7 +14,7 @@ use anyhow::{anyhow, bail, ensure, Context};
 use bytemuck::cast_slice_mut;
 use drm_ffi::drm_mode_modeinfo;
 use libc::dev_t;
-use niri_config::output::Modeline;
+use niri_config::output::{Bpc, Modeline};
 use niri_config::{Config, OutputName};
 use niri_ipc::{HSyncPolarity, VSyncPolarity};
 use smithay::backend::allocator::dmabuf::Dmabuf;
@@ -70,7 +70,11 @@ use crate::render_helpers::renderer::AsGlesRenderer;
 use crate::render_helpers::{resources, shaders, RenderCtx, RenderTarget};
 use crate::utils::{get_monotonic_time, is_laptop_panel, logical_output, PanelOrientation};
 
-const SUPPORTED_COLOR_FORMATS: [Fourcc; 4] = [
+const COLOR_FORMATS: [Fourcc; 8] = [
+    Fourcc::Xrgb2101010,
+    Fourcc::Xbgr2101010,
+    Fourcc::Argb2101010,
+    Fourcc::Abgr2101010,
     Fourcc::Xrgb8888,
     Fourcc::Xbgr8888,
     Fourcc::Argb8888,
@@ -405,6 +409,8 @@ struct ConnectorProperties<'a> {
     device: &'a DrmDevice,
     connector: connector::Handle,
     properties: Vec<(property::Info, property::RawValue)>,
+    has_change: bool,
+    requests: AtomicModeReq,
 }
 
 impl Tty {
@@ -676,12 +682,27 @@ impl Tty {
                     // Apply pending gamma changes and restore our existing gamma.
                     let device = self.devices.get_mut(&node).unwrap();
                     for (crtc, surface) in device.surfaces.iter_mut() {
-                        if let Ok(props) =
+                        if let Ok(mut props) =
                             ConnectorProperties::try_new(&device.drm, surface.connector)
                         {
-                            match reset_hdr(&props) {
-                                Ok(()) => (),
-                                Err(err) => debug!("couldn't reset HDR properties: {err:?}"),
+                            if let Some(bpc) = self
+                                .config
+                                .borrow()
+                                .outputs
+                                .find(&surface.name)
+                                .and_then(|o| o.bpc)
+                            {
+                                if let Err(err) = props.set_max_bpc(bpc) {
+                                    warn!("failed to get `max bpc` property: {err}");
+                                };
+                            }
+
+                            if let Err(err) = props.reset_hdr() {
+                                warn!("failed to get HDR properties: {err}");
+                            };
+
+                            if let Err(err) = props.commit() {
+                                warn!("failed to atomically commit properties: {err}");
                             }
                         } else {
                             warn!("failed to get connector properties");
@@ -1302,18 +1323,27 @@ impl Tty {
         debug!("picking mode: {mode:?}");
 
         let mut orientation = None;
-        if let Ok(props) = ConnectorProperties::try_new(&device.drm, connector.handle()) {
-            match reset_hdr(&props) {
-                Ok(()) => (),
-                Err(err) => debug!("couldn't reset HDR properties: {err:?}"),
+        if let Ok(mut props) = ConnectorProperties::try_new(&device.drm, connector.handle()) {
+            if let Some(bpc) = config.bpc {
+                if let Err(err) = props.set_max_bpc(bpc) {
+                    warn!("failed to get `max bpc` property: {err}");
+                };
             }
 
+            if let Err(err) = props.reset_hdr() {
+                warn!("failed to get HDR properties: {err}");
+            };
+
             match get_panel_orientation(&props) {
                 Ok(x) => orientation = Some(x),
                 Err(err) => {
                     trace!("couldn't get panel orientation: {err:?}");
                 }
             }
+
+            if let Err(err) = props.commit() {
+                warn!("failed to atomically commit properties: {err}");
+            }
         } else {
             warn!("failed to get connector properties");
         };
@@ -1441,7 +1471,7 @@ impl Tty {
             None,
             device.allocator.clone(),
             GbmFramebufferExporter::new(device.gbm.clone(), device.render_node.into()),
-            SUPPORTED_COLOR_FORMATS,
+            COLOR_FORMATS,
             // This is only used to pick a good internal format, so it can use the surface's render
             // formats, even though we only ever render on the primary GPU.
             render_formats.clone(),
@@ -1471,7 +1501,7 @@ impl Tty {
                     None,
                     device.allocator.clone(),
                     GbmFramebufferExporter::new(device.gbm.clone(), device.render_node.into()),
-                    SUPPORTED_COLOR_FORMATS,
+                    COLOR_FORMATS,
                     render_formats,
                     device.drm.cursor_size(),
                     Some(device.gbm.clone()),
@@ -2194,6 +2224,17 @@ impl Tty {
                     OutputId::next()
                 });
 
+                let props = ConnectorProperties::try_new(&device.drm, connector.handle()).ok();
+                let max_bpc = props.as_ref().and_then(|p| p.find(c"max bpc").ok());
+                let max_bpc = max_bpc.and_then(|(info, value)| {
+                    info.value_type()
+                        .convert_value(*value)
+                        .as_unsigned_range()
+                        .map(|v| v as u8)
+                });
+
+                let format = surface.map(|s| s.compositor.format().to_string());
+
                 let ipc_output = niri_ipc::Output {
                     name: connector_name,
                     make: output_name.make.unwrap_or_else(|| "Unknown".into()),
@@ -2206,6 +2247,8 @@ impl Tty {
                     vrr_supported,
                     vrr_enabled,
                     logical,
+                    max_bpc,
+                    format,
                 };
 
                 ipc_outputs.insert(id, ipc_output);
@@ -2422,6 +2465,21 @@ impl Tty {
                     },
                 };
 
+                if let Some(bpc) = config.bpc {
+                    if let Ok(mut props) =
+                        ConnectorProperties::try_new(&device.drm, surface.connector)
+                    {
+                        if let Err(err) = props.set_max_bpc(bpc) {
+                            warn!("failed to get `max bpc` property: {err}");
+                        };
+                        if let Err(err) = props.commit() {
+                            warn!("failed to atomically commit properties: {err}");
+                        }
+                    } else {
+                        warn!("failed to get connector properties");
+                    }
+                }
+
                 let change_mode = surface.compositor.pending_mode() != mode;
 
                 let vrr_enabled = surface.compositor.vrr_enabled();
@@ -3250,6 +3308,8 @@ impl<'a> ConnectorProperties<'a> {
             device,
             connector,
             properties,
+            has_change: false,
+            requests: AtomicModeReq::new(),
         })
     }
 
@@ -3262,35 +3322,71 @@ impl<'a> ConnectorProperties<'a> {
 
         Err(anyhow!("couldn't find property: {name:?}"))
     }
-}
 
-const DRM_MODE_COLORIMETRY_DEFAULT: u64 = 0;
+    fn reset_hdr(&mut self) -> anyhow::Result<()> {
+        const DRM_MODE_COLORIMETRY_DEFAULT: u64 = 0;
 
-fn reset_hdr(props: &ConnectorProperties) -> anyhow::Result<()> {
-    let (info, value) = props.find(c"HDR_OUTPUT_METADATA")?;
-    let property::ValueType::Blob = info.value_type() else {
-        bail!("wrong property type")
-    };
+        let (info, value) = self.find(c"HDR_OUTPUT_METADATA")?;
+
+        let property::ValueType::Blob = info.value_type() else {
+            bail!("wrong property type")
+        };
+        if *value != 0 {
+            self.requests
+                .add_raw_property(self.connector.into(), info.handle(), 0);
+            self.has_change = true;
+        }
+
+        let (info, value) = self.find(c"Colorspace")?;
+        let property::ValueType::Enum(_) = info.value_type() else {
+            bail!("wrong property type")
+        };
+        if *value != DRM_MODE_COLORIMETRY_DEFAULT {
+            self.requests.add_raw_property(
+                self.connector.into(),
+                info.handle(),
+                DRM_MODE_COLORIMETRY_DEFAULT,
+            );
+            self.has_change = true;
+        }
 
-    if *value != 0 {
-        props
-            .device
-            .set_property(props.connector, info.handle(), 0)
-            .context("error setting property")?;
+        Ok(())
     }
 
-    let (info, value) = props.find(c"Colorspace")?;
-    let property::ValueType::Enum(_) = info.value_type() else {
-        bail!("wrong property type")
-    };
-    if *value != DRM_MODE_COLORIMETRY_DEFAULT {
-        props
-            .device
-            .set_property(props.connector, info.handle(), DRM_MODE_COLORIMETRY_DEFAULT)
-            .context("error setting property")?;
+    fn set_max_bpc(&mut self, bpc: Bpc) -> anyhow::Result<u64> {
+        let (info, value) = self.find(c"max bpc")?;
+
+        let property::ValueType::UnsignedRange(min, max) = info.value_type() else {
+            bail!("wrong property type")
+        };
+
+        let bpc = bpc as u64;
+        let bpc = bpc.clamp(min, max);
+
+        let property::Value::UnsignedRange(value) = info.value_type().convert_value(*value) else {
+            bail!("wrong property type")
+        };
+
+        if value != bpc {
+            self.requests.add_raw_property(
+                self.connector.into(),
+                info.handle(),
+                property::Value::UnsignedRange(bpc).into(),
+            );
+            self.has_change = true;
+        }
+
+        Ok(bpc)
     }
 
-    Ok(())
+    fn commit(self) -> anyhow::Result<()> {
+        if self.has_change {
+            self.device
+                .atomic_commit(AtomicCommitFlags::ALLOW_MODESET, self.requests)?;
+        }
+
+        Ok(())
+    }
 }
 
 fn is_vrr_capable(device: &DrmDevice, connector: connector::Handle) -> Option<bool> {
diff --git a/src/backend/winit.rs b/src/backend/winit.rs
index a9bcc515..7a806cf7 100644
--- a/src/backend/winit.rs
+++ b/src/backend/winit.rs
@@ -95,6 +95,8 @@ impl Winit {
                 vrr_supported: false,
                 vrr_enabled: false,
                 logical: Some(logical_output(&output)),
+                max_bpc: None,
+                format: None,
             },
         )])));
 
diff --git a/src/ipc/client.rs b/src/ipc/client.rs
index 40a58efa..bbd05cca 100644
--- a/src/ipc/client.rs
+++ b/src/ipc/client.rs
@@ -568,6 +568,8 @@ fn print_output(output: Output) -> anyhow::Result<()> {
         vrr_supported,
         vrr_enabled,
         logical,
+        max_bpc,
+        format,
     } = output;
 
     let serial = serial.as_deref().unwrap_or("Unknown");
@@ -651,6 +653,14 @@ fn print_output(output: Output) -> anyhow::Result<()> {
         println!("  Transform: {transform}");
     }
 
+    if let Some(max_bpc) = max_bpc {
+        println!("  Max bits per channel: {max_bpc}");
+    }
+
+    if let Some(format) = format {
+        println!("  Pixel format: {format}");
+    }
+
     println!("  Available modes:");
     for (idx, mode) in modes.into_iter().enumerate() {
         let Mode {
