From 6235a690bb40e9cd206dd90ffe7db6cdc2ebf826 Mon Sep 17 00:00:00 2001
From: ElXreno <elxreno@gmail.com>
Date: Mon, 1 Dec 2025 05:08:22 +0300
Subject: [PATCH] SystemStatService: Add acpitz for temp cpu sensors list

---
 Services/System/SystemStatService.qml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Services/System/SystemStatService.qml b/Services/System/SystemStatService.qml
index 70238812..d7960480 100644
--- a/Services/System/SystemStatService.qml
+++ b/Services/System/SystemStatService.qml
@@ -37,7 +37,7 @@ Singleton {
   property real prevTime: 0
 
   // Cpu temperature is the most complex
-  readonly property var supportedTempCpuSensorNames: ["coretemp", "k10temp", "zenpower"]
+  readonly property var supportedTempCpuSensorNames: ["coretemp", "k10temp", "zenpower", "acpitz"]
   property string cpuTempSensorName: ""
   property string cpuTempHwmonPath: ""
   // For Intel coretemp averaging of all cores/sensors
@@ -429,7 +429,7 @@ Singleton {
   function updateCpuTemperature() {
     // For AMD sensors (k10temp and zenpower), only use Tctl sensor
     // temp1_input corresponds to Tctl (Temperature Control) on these sensors
-    if (root.cpuTempSensorName === "k10temp" || root.cpuTempSensorName === "zenpower") {
+    if (root.cpuTempSensorName === "k10temp" || root.cpuTempSensorName === "zenpower" || root.cpuTempSensorName === "acpitz") {
       cpuTempReader.path = `${root.cpuTempHwmonPath}/temp1_input`;
       cpuTempReader.reload();
     } // For Intel coretemp, start averaging all available sensors/cores
-- 
2.51.2

